### PR TITLE
Simplified API with 3 GET endpoints

### PR DESCRIPTION
**Pull request by seanglynn-thrive**
- Added FastAPI routers with paginated endpoint.

- Added Dummy data generator: `./bin/generate_userbehaviour_rows.py`

- Added the following API routes for code tests:
  ```
  /user_behaviour/json
  /user_behaviour/dodgy_endpoint
  /user_behaviour_10k/json
  ```
- Created inital test cases in `./tests/interview_tests/*` dir

### CODE REVIEW

**Review by Timfrazer — CHANGES_REQUESTED** [review]

File: app/routes/behaviour.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    behaviour_tbl,
+    database,
+)
+
+behaviour_router = DatabasesCRUDRouter(
+    schema=Behaviour,
+    create_schema=Behaviour,
+    table=behaviour_tbl,
+    database=database,
+    # get_all_route= False, # Default id handler doesnt match our pydantic modelling -> Set false
```
[the remark this answers is not in the document]
Comment (Timfrazer) [reply app/routes/behaviour.py:17]: Just a few minor nitpitck

**Comment by Timfrazer** [comment]
@seanglynn-thrive  would you mind running black on this? or using pre-commit that I've already confused for our other project.
https://github.com/thriveglobal/data-fivetran-lms/blob/develop/.pre-commit-config.yaml

File: app/routes/behaviour.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (seanglynn-thrive) [reply app/routes/behaviour.py:17]: What I meant by that was the default `/{id}` given by the fast-api CRUD package did not match that of what was defined in our data model, which was `user_id` I believe. I've removed all of this for now as its not needed.

**Comment by seanglynn-thrive** [comment]
@Timfrazer , I've temporarily removed `flake8` from the pre-commit yaml:
https://github.com/seanglynn-thrive/thrivey-api/pull/3/commits/d66c7be926d12002f0ee9121ca0d0b5c22cf9870

It was complaining about some unnecessary things, like our overuse of comments in the interview tests folder. I will cleanup the project and re-enable at a later point.

**Pull request merged**